src/Commands/index.ts

```diff
@@ -20,7 +20,7 @@ export class Commands {
             icon: 'pencil',
             editorCheckCallback: (checking: boolean, editor: Editor, view: MarkdownView | MarkdownFileInfo) => {
                 // TODO Need to explore what happens if a tasks code block is rendered before the Cache has been created.
-                return createOrEdit(checking, editor, view as View, this.app, this.plugin.getTasks()!);
+                return createOrEdit(checking, editor, view as View, this.app, this.plugin.getTasks());
```

Comment:
The "!" is a [Non Null Assertion Operator](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#non-null-assertion-operator-postfix-).

Basically, for whatever Reason, the Plugin can't Guarantee to Provide a Cache.
So there is a Chance that getTasks(), returns "Undefined".
The "!" Operator tells the Compiler to just ignore everything about it, and Pass it anyway.
Usually this doesn't cause a Problem, as the Cache will be there.

But what happens if it will be Undefined?
In the Worst Case, the Application Will Crash (At least in other Languages I am more Proficient in), because all attribute/function calls on the "undefined" Object in the Code will not be Callable.

Even worse, with the "!" Operator in place, I tell Every other Programmer after that Point, that just Uses the Function "createOrEdit", that allTasks, will always be there....
And a Developer that only sees that function, will never do a Null Check on it.
Causing the Potential to Crash.
This somewhat counteracts the Goal of Using Typescript in the First Place, in my Opinion.

That's why I Removed it, and routed the "Optional" Arguments via "?" true to the function that need it, performing Null Checks there, when needed.